Add cdk resources for tiles generation (#3)
opensearch-project/maps: tiles-generation/cdk/README.md

> # Tiles generation pipeline with CDK

Author: Sure, will update.

 This project in `tilesGenerationPipeline` folder helps to build a raster map tiles generaion pipeline on Amazon ECS in AWS CDK. The docker base image is come from https://github.com/Overv/openstreetmap-tile-server
.
#  Getting Started

## Prerequisites

> Install AWS CLI and Node.js first.

Reviewer: A link to how to install will be helpful

Author: Sure, will update.

## AWS Prerequisites
Have an AWS account and administrator permissions to this account. Use `aws configure` to set account and region.

## Update configuration/performance tunning

> The tiles generation configuration could be updated at `cdk.ts`. For example, Users can custom tiles generation area, zoom leve, schedule generation, etc.

Reviewer: ```suggestion
The tiles generation configuration could be updated at `cdk.ts`. For example, Users can custom tiles generation area, zoom level, schedule generation, etc.
```

## Deploy test tile generation stack

> It's highly recommended to test the tile generation process with light data before running whole planet.

Reviewer: I don’t think it is required .

* Replace `YourS3BucketName` to your bucket name to receive generated tiles. Confirm your S3 bucket name passed into CDK
```
#Pass S3 Bucket name to CDK.
cdk synth --context TILE_S3_BUCKET=YourS3BucketName TestTileGenerationStack
```

> * Deploy stack

Reviewer: What does stack mean?

```
cdk deploy TestTileGenerationStack -c TILE_S3_BUCKET=YourS3BucketName
```

* Check test stack result
From AWS console to check ECS task status, after checking test ECS task finished exit with 0, disable the ECS task schedule and change ASG EC2 instance desiredCapacity to 0 to save resources.

## Deploy planet tile generation stack

* Replace `YourS3BucketName` to your bucket name to receive generated tiles. Confirm your S3 bucket name passed into CDK
```
#Pass S3 Bucket name to CDK.

> cdk synth --context TILE_S3_BUCKET=YourS3BucketName PlanetTileGenerationStack

Author: we don't need this step, we can directly pass S3 bucket name at cdk deploy.

# Tiles generation pipeline with CDK

>  This project in `tilesGenerationPipeline` folder helps to build a raster map tiles generaion pipeline on Amazon ECS in AWS CDK. The docker base image is come from https://github.com/Overv/openstreetmap-tile-server.

Reviewer 2: ```suggestion
 The tiles generation stack builds a raster map tiles generation pipeline on [Amazon ECS](https://aws.amazon.com/ecs/) using [AWS CDK](https://aws.amazon.com/cdk/). The docker image used in ECS comes from the [openstreetmap-tile-server](https://github.com/Overv/openstreetmap-tile-server) project.
```

## Diagram

> ![alt text](./tiles-generation-diagram.png).

Reviewer 2: Typo in cluster. Also, diagram inside Cloudformation is a little bit confusing. I would maybe break the diagram into numbered steps.

Author: Updated, I added steps explanation for the diagram.

#  Getting Started
## Prerequisites

### Install Node.js
Visit [here](https://nodejs.org/en/) to install Node.js

### Install AWS CDK Toolkit
```
npm install -g aws-cdk
```

### Install AWS CLI
Visit [here](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html) to install AWS CLI
### AWS Prerequisites
Visit [here](https://docs.aws.amazon.com/cdk/latest/guide/getting_started.html#getting_started_prerequisites) to configure aws credentials. 
## Deployment

### Deploy steps

1. (Optional)The tiles generation configuration could be customized at `cdk.ts`. For example, Users can customize tiles generation area, zoom level, schedule generation, etc.

> 2. A S3 bucket is required in CDK deployment to store generated tiles. The context name `TILE_S3_BUCKET` you will use to deploy. This bucket must be in the same aws account.

Reviewer 2: Does S3 bucket need to be created before running the CDK deployment?

Author: Yes, currently we need an existing S3 bucket for deployment. In the future, we can support create new S3 bucket directly by CDK and use it.

3. Deploy CDK stacks, all AWS resources defined within the scope of a stack

> * Test tile generation stack, it's used to test with light data for development and tuning performance.

Reviewer 2: Add more description about what light data is being used

Author: Updated

```
cdk deploy TestTileGenerationStack --context TILE_S3_BUCKET=S3BucketName
```

> * Planet tile generation stack

Reviewer 2: Add where the data comes from

Author: Updated

```
cdk deploy PlanetTileGenerationStack --context TILE_S3_BUCKET=S3BucketName
```

> 4. Execute tiles generation. Once stack deployed, you will get the outputs of `ClusterName`, `TaskDefinitionName`, `CapacityProviderName` on the terminal end. Use them on the below command.

Reviewer 2: ```suggestion
4. Execute tiles generation. Once stack deployed, note the `ClusterName`, `TaskDefinitionName`, `CapacityProviderName` from the output of the command. Use them on the below command.
```

Author: Updated

```

> aws ecs run-task --cluster ClusterName --task-definition TaskDefinitionName --capacity-provider-strategy capacityProvider=CapacityProviderName

Reviewer 2: Is ECS region indpendent?

Author: The ECS task region is same with the ECS cluster(the stack), which can be pre-configured when setting up aws credentials.
